### PR TITLE
P3/cameron/jungle server #66

### PR DESCRIPTION
**Pull request by camja014**
Created skeleton structure for the Jungle Server.

Features include the following:
- A simple server class that extends com.esotericsoftware.kryonet.Server
- Jungle server class can hand out DataSource Connection objects.
- Client Request Listener Skeletons
- Request Controller Skeletons
- Definitions for most client request, server response, and event messages.
- Ability to load server and database configurations from a Java Properties file.

Todo:
- Fill in all listener skeletons with controller calls.
- Fill in all controller classes to process requests and form responses.
- Add any missing response/request/event types that we want to utilize.
- Add a number of Data Access Objects (Dao's) to interact with the database.

### CODE REVIEW

File: server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/data/Types.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        RED,
+        BLACK
+    }
+
+    public enum GameStatus {
+        WAITING,
+        IN_PROGRESS,
+        DRAW,
+        P1_WINNER,
+        P2_WINNER
```
Comment (tking2096) [inline server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/data/Types.java]: Is this redundant with the GameOutcomeType? Perhaps FINISHED would suffice?

File: server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/server/Controller.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package edu.colostate.cs.cs414.chessirecoders.JungleServer.server;
+
+import com.esotericsoftware.kryonet.Connection;
+
+public abstract class Controller implements Runnable {
```
Comment (tking2096) [inline server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/server/Controller.java]: Are the classes that implement Controller in charge of delegating flow of responses? A short JavaDoc comment couldn't hurt.

**Comment by tking2096** [comment]
It would be super helpful if you could do what you can to shorten your PR's. 850 lines of code is a lot to read over ;)

**Review by ALHennings — COMMENTED** [review]
This is a lot of information in one update, but it looks good. It seems accurate for a skeleton and my limited understanding. Approved.

**Review by ALHennings — DISMISSED** [review]

**Review by ALHennings — DISMISSED** [review]

File: server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/server/Controller.java (hunk 2 above)
Comment (camja014) [reply server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/server/Controller.java]: Added in my latest push

File: server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/data/Types.java (hunk 1 above)
Comment (camja014) [reply server/src/main/java/edu/colostate/cs/cs414/chessirecoders/JungleServer/data/Types.java]: Yeah, that is redundant, which I'll fix. We still need a way to track the winner though.

Perhaps that can be another attribute of Game?

**Comment by camja014** [comment]
Alright, so I changed a bunch of stuff with the maven files to get it to build successfully in Travis:

1. Instead of having separate maven projects, it's now organized into one central project and three child modules.

2. I stopped tracking all eclipse and IntelliJ builds files (.classpath and .project). 

I don't think I did it very gracefully though, **so pulling from this will probably nuke your guy's local copies**. But this is probably going to require you to re-import the project anyways. The good news is that it's extremely easy to import the project now! 

In Eclipse, all you have to do is go to File -> Import -> Maven -> Existing Maven Project. Point it to the root directory of the project, and it should pick up all of the projects automagically. 

Then you can add a new "Maven Build" run configuration, set the "Base Directory" to the repo root directory, and set the "Goals" to "clean install" and it should build without issue (assuming you have a version of jdk_1.8 added to eclipse, as I've told maven to use 1.8 specifically)

I've tested it several times and it works great.

**Comment by camja014** [comment]
Also, sorry about the massive number of changes to the entire project...

I like having stuff be organized and I get carried away sometimes.... But hey, it builds now!

**Review by amgunden — APPROVED** [review]

**Pull request merged**